tests/drivers/adc/adc_api/src/test_adc.c

```diff
@@ -764,26 +291,20 @@ static int test_task_repeated_samplings(void)
 		/* Start consecutive samplings as fast as possible. */
 		.interval_us     = 0,
 	};
-	const struct adc_sequence sequence = {
+	struct adc_sequence sequence = {
 		.options     = &options,
-#if defined(ADC_2ND_CHANNEL_ID)
-		.channels    = BIT(ADC_1ST_CHANNEL_ID) |
-			       BIT(ADC_2ND_CHANNEL_ID),
-#else
-		.channels    = BIT(ADC_1ST_CHANNEL_ID),
-#endif /* defined(ADC_2ND_CHANNEL_ID) */
 		.buffer      = m_sample_buffer,
 		.buffer_size = sizeof(m_sample_buffer),
-		.resolution  = ADC_RESOLUTION,
 	};
 
-	const struct device *adc_dev = init_adc();
+	init_adc();
+	(void)adc_sequence_init_dt(&adc_channels[0], &sequence);
 
-	if (!adc_dev) {
-		return TC_FAIL;
+	if (adc_channels_count > 1) {
+		sequence.channels |=  BIT(adc_channels[1].channel_id);
```

Comment:
Just for consistency, it would be better to also here use:
```suggestion
	for (int i = 1; i < adc_channels_count; i++) {
		sequence.channels |= BIT(adc_channels[i].channel_id);
```
Then it should be possible to use even more than 2 channels for a board, provided that the supplied buffer is big enough (but that is probably beyond the scope of this PR).

Reply:
Agreed, wilL keep this in mind as I try to improvE this test in a future PR.